knowhere/index/VecIndexFactory.cpp

```diff
@@ -100,23 +100,24 @@ VecIndexFactory::CreateVecIndex(const IndexType& type, const IndexMode mode) {
         }
 #ifdef KNOWHERE_GPU_VERSION
         case IndexMode::MODE_GPU: {
-            auto gpu_device = -1;  // TODO: remove hardcode here, get from invoker
+            auto gpu_device = 0;  // TODO: remove hardcode here, get from invoker
+            knowhere::FaissGpuResourceMgr::GetInstance().InitDevice(0, 200 * 1024 * 1024, 300 * 1024 * 1024, 2);
             if (type == IndexEnum::INDEX_FAISS_BIN_IDMAP) {
                 return std::make_shared<knowhere::BinaryIDMAP>();
             } else if (type == IndexEnum::INDEX_FAISS_BIN_IVFFLAT) {
                 return std::make_shared<knowhere::BinaryIVF>();
             } else if (type == IndexEnum::INDEX_FAISS_IDMAP) {
                 return std::make_shared<knowhere::GPUIDMAP>(gpu_device);
             } else if (type == IndexEnum::INDEX_FAISS_IVFFLAT) {
-                return std::make_shared<knowhere::GPUIVF_NM>(gpu_device);
+                return std::make_shared<knowhere::GPUIVF>(gpu_device);
```

Comment:
need use GPUIVF_NM, not GPUIVF

Reply:
> need use GPUIVF_NM, not GPUIVF

GPUIVF_NM load function not realized.

Reply:
I have tested GPUIVF_NM in my machine, it can pass all UT

Reply:
> I have tested GPUIVF_NM in my machine, it can pass all UT
when used in milvus will fail, see IndexGPUIVF.cpp:66